drivers/modem/modem_cellular.c

```diff
@@ -0,0 +1,1167 @@
+		modem_cellular_delegate_event(data, MODEM_CELLULAR_EVENT_SCRIPT_SUCCESS);
+	} else {
+		modem_cellular_delegate_event(data, MODEM_CELLULAR_EVENT_SCRIPT_FAILED);
+	}
+}
+
+static void modem_cellular_chat_on_imei(struct modem_chat *chat, char **argv, uint16_t argc,
+					void *user_data)
+{
+	struct modem_cellular_data *data = (struct modem_cellular_data *)user_data;
+
+	if (argc != 2) {
+		return;
+	}
+
+	if (strlen(argv[1]) != 15) {
+		return;
+	}
+
+	for (uint8_t i = 0; i < 15; i++) {
+		data->imei[i] = argv[1][i] - '0';
+	}
+}
+
+static void modem_cellular_chat_on_cgmm(struct modem_chat *chat, char **argv, uint16_t argc,
+					void *user_data)
+{
+	struct modem_cellular_data *data = (struct modem_cellular_data *)user_data;
+
+	if (argc != 2) {
+		return;
+	}
+
+	strncpy(data->hwinfo, argv[1], sizeof(data->hwinfo) - 1);
+}
+
+static void modem_cellular_chat_on_creg(struct modem_chat *chat, char **argv, uint16_t argc,
+					void *user_data)
+{
+	struct modem_cellular_data *data = (struct modem_cellular_data *)user_data;
+
+	if (argc != 3) {
+		return;
+	}
+
+	data->access_tech = atoi(argv[1]);
+	data->registration_status = atoi(argv[2]);
+}
+
+static void modem_cellular_chat_on_cgatt(struct modem_chat *chat, char **argv, uint16_t argc,
+					 void *user_data)
+{
+	struct modem_cellular_data *data = (struct modem_cellular_data *)user_data;
+
+	if (argc != 2) {
+		return;
+	}
+
+	data->packet_service_attached = atoi(argv[1]);
+}
+
+MODEM_CHAT_MATCH_DEFINE(ok_match, "OK", "", NULL);
+MODEM_CHAT_MATCH_DEFINE(imei_match, "", "", modem_cellular_chat_on_imei);
+MODEM_CHAT_MATCH_DEFINE(cgmm_match, "", "", modem_cellular_chat_on_cgmm);
+MODEM_CHAT_MATCH_DEFINE(creg_match, "+CREG: ", ",", modem_cellular_chat_on_creg);
+MODEM_CHAT_MATCH_DEFINE(cgatt_match, "+CGATT: ", ",", modem_cellular_chat_on_cgatt);
+
+MODEM_CHAT_MATCHES_DEFINE(abort_matches, MODEM_CHAT_MATCH("ERROR", "", NULL));
+
+MODEM_CHAT_MATCHES_DEFINE(connect_abort_matches,
+			  MODEM_CHAT_MATCH("ERROR", "", NULL),
+			  MODEM_CHAT_MATCH("BUSY", "", NULL),
+			  MODEM_CHAT_MATCH("NO ANSWER", "", NULL),
+			  MODEM_CHAT_MATCH("NO CARRIER", "", NULL),
+			  MODEM_CHAT_MATCH("NO DIALTONE", "", NULL));
+
+MODEM_CHAT_SCRIPT_CMDS_DEFINE(init_chat_script_cmds,
+			      MODEM_CHAT_SCRIPT_CMD_RESP_NONE("AT", 100),
+			      MODEM_CHAT_SCRIPT_CMD_RESP_NONE("AT", 100),
+			      MODEM_CHAT_SCRIPT_CMD_RESP_NONE("AT", 100),
+			      MODEM_CHAT_SCRIPT_CMD_RESP_NONE("AT", 100),
+			      MODEM_CHAT_SCRIPT_CMD_RESP("ATE0", ok_match),
+			      MODEM_CHAT_SCRIPT_CMD_RESP("AT+CMEE=1", ok_match),
+			      MODEM_CHAT_SCRIPT_CMD_RESP("AT+CREG=0", ok_match),
+			      MODEM_CHAT_SCRIPT_CMD_RESP("AT+CGSN", imei_match),
+			      MODEM_CHAT_SCRIPT_CMD_RESP("", ok_match),
+			      MODEM_CHAT_SCRIPT_CMD_RESP("AT+CGMM", cgmm_match),
+			      MODEM_CHAT_SCRIPT_CMD_RESP("", ok_match),
+			      MODEM_CHAT_SCRIPT_CMD_RESP("AT+CMUX=0,0,5,127,10,3,30,10,2",
+							 ok_match));
```

Comment:
fyi: I observed that some modems do not reply to a successful AT+CMUX command with "OK".

Reply:
I updated the driver to not wait for the OK match and instead wait for a timeout of 100ms, this works for the Quectel modems which do send an OK as well